app/models/measurement_details.rb

```diff
@@ -21,12 +21,17 @@ def initialize(attributes = {})
   def load
     validate!
 
-    load_total_from_gr
-    load_local_from_gr
-    load_user_from_gr
-    load_sub_mins_from_gr
-    load_team_from_gr
-    load_split_measurements
+    # make sure instance vars are set
+    @current_power = Power.current
+    gr_singleton
+    ministry
+
+    threads = [:load_local_from_gr, :load_total_from_gr, :load_user_from_gr, :load_sub_mins_from_gr,
+               :load_team_from_gr, :load_split_measurements]
+    threads = threads.map do |method|
+      Thread.new { send(method) }
+    end
+    threads.each { |t| t.join(1) }
```

Comment:
Why do you give a 1 second limit? If for some reason the requests take longer than 1 second, could that cause the data to be loaded incorrectly?